src/TLuaInterpreter.cpp

```diff
@@ -8131,7 +8131,10 @@ int TLuaInterpreter::getMapLabel(lua_State* L)
     }
 
     Host& host = getHostFromLua(L);
-    auto pA = host.mpMap->mpRoomDB->getArea(area);
+    auto pA = host.mpMap->mpRoomDB->getArea(areaId);
+    if (!pA) {
+        return warnArgumentValue(L, __func__, QStringLiteral("areaID %1 does not exist").arg(areaId));
```

Comment:
This type of error is not a wrong argument one - the arguments are given in an OK mannger - but a nil+error one.

Reply:
Doesn't `warnArgumentValue(...)` cover this sort of situation - an integer was supplied as an areaID but that area does not exist. That surely is a run-time eror - i.e. a nil + error message case warning that the value supplied does not match an extant area?

Reply:
Yeah you are totally right.